src/lib.rs

```diff
@@ -1531,10 +1529,29 @@ fn encode_block(seq: &Sequence, fi: &FrameInvariants, fs: &mut FrameState,
         TxType::DCT_DCT
     };
 
-    // if inter, predict here
-    //mode.predict(&mut rec.mut_slice(po), tx_size);
-
     if is_inter {
+        // Inter mode prediction can take place once for a whole partition,
+        // instead of each tx-block.
+        for p in 0..3 {
```

Comment:
btw, I found that I don't need iterate over planes here, since write_tx_tree() take care of all planes by itself, and it also check whether there is chroma partition exist.

Reply:
Oh- no, the  luma_mode.predict_inter() should be still called for each plane, just it sholud not be called if there is no chroma block.